appwindow.js

```diff
@@ -107,6 +107,24 @@ var AppWindow = GObject.registerClass(
             this.method_handler(this.settings, 'changed::window-position', this.update_resize_boxes);
             this.update_resize_boxes();
 
+            const HEIGHT_MOD = 0.05;
+            this.simple_action('window-size-dec', () => {
+                if (this.settings.get_boolean('window-maximize'))
+                    this.settings.set_double('window-size', 1.0);
```

Comment:
You're calling `settings.set_double` twice: here, then in `adjust_double_setting` again. It triggers all callbacks subscribed to `changed::window-size`. Currently, they do nothing after the first `set_double` call, but it might change. Please change the code to call `set_double` only once (i. e. `adjust_double_setting` should not be called if `window-maximize` is `true`)